opm/autodiff/BlackoilModelBase.hpp

```diff
@@ -92,24 +93,21 @@ namespace Opm {
             ADB              mob;   // Phase mobility (per cell)
         };
 
-        struct SimulatorData {
+        struct SimulatorData : public Opm::FIPDataEnums {
             SimulatorData(int num_phases);
 
-            enum FipId {
-                FIP_AQUA = Opm::Water,
-                FIP_LIQUID = Opm::Oil,
-                FIP_VAPOUR = Opm::Gas,
-                FIP_DISSOLVED_GAS = 3,
-                FIP_VAPORIZED_OIL = 4,
-                FIP_PV = 5,                    //< Pore volume
-                FIP_WEIGHTED_PRESSURE = 6
-            };
+            using Opm::FIPDataEnums :: FipId ;
```

Comment:
The problem is that in some places of the code SimulatorData::FIP_WEIGHTED_PRESSURE is used. This is ok since SimulatorData is inherited. I'm not saying that this is the only way to solve this but I'm on the other hand not convinced that you fully understood the dimension of your request. That is why I said please provide a PR to this PR that solves the problem in the way you feel good about it.